pkg/ds/farm.go

```diff
@@ -262,8 +262,13 @@ func (dsf *Farm) handleDSChanges(changes dsstore.WatchedDaemonSets, quitCh <-cha
 		return
 	}
 
+	if changes.Total == 0 {
+		dsf.logger.Errorf("No daemon sets present, asked to delete %d; farm refuses to delete its %d daemon sets", len(changes.Deleted), len(dsf.children))
```

Comment:
Would you prefer (as I am suspecting) simply having the watch not report such a situation, as #751 does?

Aborting the entire farm could be a choice. Can I be sure I'm not going to hammer Consul by repeatedly asking for more watches if this happens repeatedly for a period of time? Maybe have to back off for a bit.